sw/device/tests/sim_dv/keymgr_key_derivation.c

```diff
@@ -0,0 +1,278 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/dif/dif_kmac.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+// TODO, change to use dif_flash_ctrl.h
+#include "sw/device/lib/flash_ctrl.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+#include "sw/device/silicon_creator/lib/base/sec_mmio.h"
```

Comment:
We should use the difs for keymgr and lifecycle. The original test was focusing on testing the rom drivers, and that's why we were using the `silicon_creator/lib` dependencies. However, this is working, so not sure it needs to change.

@timothytrippel any thoughts?

Reply:
Yes that's correct, we should use the DIFs. They are probably more efficient for testing too (less checks).

Reply:
@timothytrippel should we switch to use dif_keymgr? should we also remove `silicon_creator/lib/drivers/keymgr.c`, so that all tests use the dif?